### PR TITLE
feat(common): add receipts type

### LINKED ISSUE #1721: Add receipts to common data types
Required in order for p2p sync to not get coupled to gateway types.

### PR DESCRIPTION
**Pull request by Mirko-von-Leipzig**
Title.

Required for p2p sync to not rely on gateway types.

Closes #1721

### CODE REVIEW

File: crates/common/src/receipt.rs [hunk 1]
```diff
@@ -0,0 +1,60 @@
+use crate::prelude::*;
+
+#[derive(Clone, Default, Debug, PartialEq)]
+pub struct Receipt {
+    pub actual_fee: Option<Fee>,
+    pub events: Vec<crate::event::Event>,
+    pub execution_resources: Option<ExecutionResources>,
+    pub l1_to_l2_consumed_message: Option<L1ToL2Message>,
```
Comment (CHr15F0x) [inline crates/common/src/receipt.rs:8]: I think this field is redundant with the respective transaction which should be of type L1_HANDLER.
```suggestion
```

File: crates/common/src/receipt.rs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct L1ToL2Message {
+    pub from_address: EthereumAddress,
+    pub payload: Vec<L1ToL2MessagePayloadElem>,
+    pub selector: EntryPoint,
+    pub to_address: ContractAddress,
+    pub nonce: Option<L1ToL2MessageNonce>,
+}
```
Comment (CHr15F0x) [inline crates/common/src/receipt.rs:22]:
```suggestion
```

File: crates/common/src/receipt.rs (hunk 1 above)
Comment (CHr15F0x) [reply crates/common/src/receipt.rs:8]: For this reason this field is not present in the rpc.

**Review by CHr15F0x — APPROVED** [review]
LGTM :+1:
(mod: pls remove the L1ToL2Message)

File: crates/common/src/receipt.rs (hunk 1 above)
Comment (kkovaacs) [reply crates/common/src/receipt.rs:8]: Yep, this seems redundant with the data we already have in the transaction.

**Pull request merged**